subprojects/launcher/src/main/java/org/gradle/launcher/daemon/configuration/BuildProcess.java

```diff
@@ -43,9 +43,19 @@ public boolean configureForBuild(DaemonParameters requiredBuildParameters) {
 
         final JvmOptions jvmOptions = new JvmOptions(new IdentityFileResolver());
         jvmOptions.systemProperties(getJvmOptions().getImmutableSystemProperties());
+        if (requiredBuildParameters.hasUserConfiguredImmutableJvmArgs()) {
```

Comment:
In case the user did not define any args, we want to ignore the current immutables (case 1 in described in #2134). 

This test would fail for example: https://github.com/gradle/gradle/pull/2136/files#diff-2dc6943d82d41c3c807602f5690618e1R130